Feature/scrum 3595

tlm opened the pull request.

Commit:
Start of README for the project.

Commit:
Added licensing information.

Commit:
Started initial maven project.

Commit:
Start of spring-boot application setup.

Commit:
Changed spring annotation for entry point

Commit:
Added Apache HTTP async client lib.

Commit:
Reconfigured Application start.

Also added the start of work for an RDAP client.

Commit:
Added route handlers for RDAP path segments

Commit:
Fixed README for GitHub.

Was not handling new lines for links.

Commit:
Adding ip path segment handler.

This path segment is a bit weird to implement in spring as it try's to tokenise
everything on '/' and no way to turn it off. Future changes may try to install
a custom path handler into spring.

Commit:
Documentation for PathRestController

Commit:
More documentation and HEAD request support.

jimvella commented:
Controller is a brittle mechanism to implement the pass though proxy. Implementing a Filter (https://docs.oracle.com/javaee/7/api/javax/servlet/Filter.html) instead will decouple the pass though functionality from any knowledge of rdap. The pass though mechanism should preserve as much of the original request as possible, since the proxied app may actually want this stuff (i.e. last-modified, etag, etc).

Edit: It might be worth building on top of someone else's proxy 
* https://github.com/mitre/HTTP-Proxy-Servlet 
* https://github.com/adamfisk/LittleProxy

Commit:
Fixed bug where controller was using wrong method

Commit:
Fixed issue with wrong method for handler

Commit:
Removed original code and added zuul.

Have removed the initial implementation of rest controllers and added Netflix
Zuul.

Further filters will be implemented that perform the routing for the project.

Commit:
Removed unused dependencies.
